docs/user/configuring-agent.md

```diff
@@ -0,0 +1,131 @@
+| `system-info` | `array` (`string`) | | System info that the agent shall include in status updates from built-in collectors. See [Built-in system info collectors](#built-in-system-info-collectors). Default: `[]` |
+| `system-info-custom` | `array` (`string`) | | System info that the agent shall include in status updates from user-defined collectors. See [Custom system info collectors](#custom-system-info-collectors). Default: `[]` |
+| `system-info-timeout` | `Duration` | | The timeout for collecting system info. Default: `2m` |
+| `log-level` | `string` | | The level of logging: "panic", "fatal", "error", "warn"/"warning", "info", "debug", or "trace". Default: `info` |
+
+`Duration` values are strings of an integer value with appended unit of time ('s' for seconds, 'm' for minutes, or 'h' for hours). Examples: `30s`, `10m`, `24h`
+
+## Built-in system info collectors
+
+The agent has a set of built-in collectors for system information. You can see the information collected by these collectors using the following command:
+
+```console
+sudo flightctl-agent system-info | jq '.'
+```
+
+Out of these, the agent includes a standard set of system infos in its device status:
+
+```console
+status:
+  [...]
+  systemInfo:
+    architecture: amd64
+    operatingSystem: linux
+    agentVersion: v0.7.0
+    bootID: 87f7e27e-bdc0-42b1-b909-6dc81fe43ea2
+```
+
+You can specify extra system infos to be included in the device status by listing them under the `system-info` configuration parameter:
+
+| Extra System Info |
+| ----------------- |
+| `hostname` |
+| `kernel` |
+| `distroName` |
+| `distroVersion` |
+| `productName` |
+| `productSerial` |
+| `productUuid` |
+| `biosVendor` |
+| `biosVersion` |
+| `netInterfaceDefault` |
+| `netIpDefault` |
+| `netMacDefault` |
+| `gpu` |
+| `memoryTotalKb` |
+| `cpuCores` |
+| `cpuProcessors` |
+| `cpuModel` |
+
+For example, if you add the following parameter to your agent's `config.yaml`
+
+```console
+system-info: [hostname, kernel, distroName, distroVersion]
+```
+
+then the reported device status might look like
+
+```console
+status:
+  [...]
+  systemInfo:
+    architecture: amd64
+    operatingSystem: linux
+    agentVersion: v0.7.0
+    bootID: 87f7e27e-bdc0-42b1-b909-6dc81fe43ea2
+    hostname: device.example.com
+    kernel: 5.14.0-503.38.1.el9_5.x86_64
+    distroName: Red Hat Enterprise Linux
+    distroVersion: 9.5 (Plow)
+```
+
+## Custom system info collectors
+
+You can specify custom system info collectors that the agent calls and whose output it includes under `.status.systemInfo.customInfo` in the device status.
+
+To add a key `myInfo`,
+
+1. add an executable with that name to `/usr/lib/flightctl/custom-info.d/` that when it is executed returns the desired value, and
+2. enable the collection and reporting of this info by adding the key `myInfo` to the agent's `config.yaml` under the `system-info-custom` configuration parameter.
```

Comment:
we do support myinfo 01-myinfo and 01-myinfo.{sh,pl,otherthing}. I just wonder if the camelCase name would be awkward, although logical.